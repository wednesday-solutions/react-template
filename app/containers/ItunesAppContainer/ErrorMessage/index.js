/**
 *
 * ErrorMessage
 *
 */

import React from 'react';
// import PropTypes from 'prop-types'
// import styled from 'styled-components'
import { Result, Button } from 'antd';

import { FormattedMessage as T, injectIntl } from 'react-intl';

function ErrorMessage({ intl: { formatMessage } }) {
  return (
    <div data-testid="error-message">
      <Result
        status="warning"
        title={formatMessage({ id: 'error_message' })}
        extra={
          <Button type="primary" key="console">
            <T id="reload" />
          </Button>
        }
      />
    </div>
  );
}

export default injectIntl(ErrorMessage);
