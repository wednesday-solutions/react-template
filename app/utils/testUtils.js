import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import { IntlGlobalProvider } from '@components/IntlGlobalProvider';
import { theme } from '@containers/App';

export const renderWithIntl = (children) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children, history) => {
  const store = configureStore({}, browserHistory).store;
  return render(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          {history ? <Router history={history}>{children}</Router> : <BrowserRouter>{children}</BrowserRouter>}
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export const getStyles = () => {};
