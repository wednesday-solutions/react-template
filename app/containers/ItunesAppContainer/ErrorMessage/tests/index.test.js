/**
 *
 * Tests for ErrorMessage
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import ErrorMessage from '../index';

describe('<ErrorMessage />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<ErrorMessage />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 ErrorMessage component', () => {
    const { getAllByTestId } = renderWithIntl(<ErrorMessage />);
    expect(getAllByTestId('error-message').length).toBe(1);
  });
});
