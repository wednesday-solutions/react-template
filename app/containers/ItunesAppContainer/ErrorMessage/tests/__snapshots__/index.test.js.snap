// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      data-testid="error-message"
    >
      <div
        class="ant-result ant-result-warning"
      >
        <div
          class="ant-result-icon"
        >
          <span
            aria-label="warning"
            class="anticon anticon-warning"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="warning"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M955.7 856l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zM480 416c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v184c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V416zm32 352a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
              />
            </svg>
          </span>
        </div>
        <div
          class="ant-result-title"
        >
          There are some problems with your operation.
        </div>
        <div
          class="ant-result-extra"
        >
          <button
            class="ant-btn ant-btn-primary"
            type="button"
          >
            Reload
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
